Assets/MixedRealityToolkit/MSBuild/Scripts/ReferenceItemInfo.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// A common base class for reference items such as C# Projects and DLLs to be added to MSBuild.
+    /// </summary>
+    public class ReferenceItemInfo
+    {
+        protected readonly IEnumerable<CompilationPlatformInfo> availablePlatforms;
+
+        /// <summary>
+        /// Gets the Guid associated with the reference.
+        /// </summary>
+        public Guid Guid { get; }
+
+        /// <summary>
+        /// Gets the output path to the reference.
+        /// </summary>
+        public Uri ReferencePath { get; }
+
+        /// <summary>
+        /// Gets name of the reference item.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <remarks>
+        /// In the editor, we can support all patforms if it's a pre-defined assembly, or an asmdef with Editor platform checked. 
+        /// Otherwise we fallback to just the platforms specified in the editor.
+        /// </remarks>
+        public IReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> InEditorPlatforms { get; protected set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <remarks>
+        /// In the player, we support any platform if pre-defined assembly, or the ones explicitly specified in the AsmDef player.
```

Comment:
Nit: It's weird seeing the summaries empty here, but having remarks (i.e. clarifying comments, but no main comments to clarify)